src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -54,84 +55,50 @@ protected void generateFailureSuite(XmlSuite xmlSuite, ISuite suite, String outp
     failedSuite.setName("Failed suite [" + xmlSuite.getName() + "]");
     m_xmlSuite= failedSuite;
 
-    Map<String, XmlTest> xmlTests= Maps.newHashMap();
-    for(XmlTest xmlT: xmlSuite.getTests()) {
-      xmlTests.put(xmlT.getName(), xmlT);
-    }
-
     Map<String, ISuiteResult> results = suite.getResults();
 
-    synchronized(results) {
```

Comment:
Ok, I think you're right. I had #1168 in mind but `results` is not changed, so it should not fail :crossed_fingers: